lib/liquid/tags/render.rb

```diff
@@ -38,13 +41,25 @@ def render_tag(context, output)
         parse_context: parse_context
       )
 
-      inner_context = context.new_isolated_subcontext
-      inner_context.template_name = template_name
-      inner_context.partial = true
-      @attributes.each do |key, value|
-        inner_context[key] = context.evaluate(value)
+      context_variable_name = @alias_name || template_name.split('/').last
+
+      variable = if @variable_name_expr
+        context.evaluate(@variable_name_expr)
+      else
+        context.find_variable(template_name, raise_on_not_found: false)
```

Comment:
Does this automatically pass a variable whose name is the same as the snippet name? I wouldn't keep that behavior. If there is no `with` or `for`, we should pass nothing down.

Reply:
I've got a change ready to set this to `nil`. It was ignoring this down further by checking if `with` was set